tests/epyccel/recognised_functions/test_math_funcs.py

```diff
@@ -377,12 +382,735 @@ def floor_return_type_real(x):
+    def expm1_type(x):
+        from math import expm1
+        return expm1(x)
+
+    f1 = epyccel(expm1_type, language = language)
+    x = uniform(high = 700.0)
+
+    assert(type(expm1_type(x)) == type(f1(x)))
+    assert(type(expm1_type(-x)) == type(f1(-x)))
+
+#--------------------------- log1p function ------------------------------#
+
+def test_log1p_call(language = 'c'):
+    @types('real')
+    def log1p_call(x):
+        from math import log1p
+        return log1p(x)
+
+    f1 = epyccel(log1p_call, language = language)
+    x = rand()
+    assert(isclose(f1(x) ,  log1p_call(x), rtol=1e-15, atol=1e-15))
+    assert isinstance(f1(x), type(log1p_call(x)))
+
+def test_log1p_phrase(language = 'c'):
+    @types('real','real')
+    def log1p_phrase(x,y):
+        from math import log1p
+        a = log1p(x)+log1p(y)
+        return a
+
+    f2 = epyccel(log1p_phrase, language = language)
+    x = rand()
+    y = rand()
+    assert(isclose(f2(x,y) ,  log1p_phrase(x,y), rtol=1e-15, atol=1e-15))
+
+#--------------------------- log2 function ------------------------------#
+
+def test_log2_call(language = 'c'):
+    @types('real')
+    def log2_call(x):
+        from math import log2
+        return log2(x)
+
+    f1 = epyccel(log2_call, language = language)
+    low = min_float
+    high = max_float
+    x = uniform(low=low, high=high)
+    assert(isclose(f1(x) ,  log2_call(x), rtol=1e-15, atol=1e-15))
+    assert isinstance(f1(x), type(log2_call(x)))
+
+def test_log2_phrase(language = 'c'):
+    @types('real','real')
+    def log2_phrase(x,y):
+        from math import log2
+        a = log2(x)+log2(y)
+        return a
+
+    f2 = epyccel(log2_phrase, language = language)
+    low = min_float
+    high = max_float
+    x = uniform(low=low, high=high)
+    y = uniform(low=low, high=high)
+    assert(isclose(f2(x,y) ,  log2_phrase(x,y), rtol=1e-15, atol=1e-15))
+
+#--------------------------- log10 function ------------------------------#
+
+def test_log10_call(language):
+    @types('real')
+    def log10_call(x):
+        from math import log10
+        return log10(x)
+
+    f1 = epyccel(log10_call, language = language)
+    low = min_float
+    high = max_float
+    x = uniform(low=low, high=high)
+    assert(isclose(f1(x) ,  log10_call(x), rtol=1e-15, atol=1e-15))
+    assert isinstance(f1(x), type(log10_call(x)))
+
+def test_log10_phrase(language):
+    @types('real','real')
+    def log10_phrase(x,y):
+        from math import log10
+        a = log10(x)+log10(y)
+        return a
+
+    f2 = epyccel(log10_phrase, language = language)
+    low = min_float
+    high = max_float
+    x = uniform(low=low, high=high)
+    y = uniform(low=low, high=high)
+    assert(isclose(f2(x,y) ,  log10_phrase(x,y), rtol=1e-15, atol=1e-15))
+
+#--------------------------------- Pow function ------------------------------#
+
+def test_pow_call(language = 'c'):
+    @types('real', 'real')
+    def pow_call(x, y):
+        from math import pow
```

Comment:
Use `from math import pow as my_pow` to avoid Codacy "Redefining built-in" warning